### ISSUE TITLE
Heex : this snippet triggers a hard crash

### ISSUE THREAD

**Issue by ArthurMmn** [zed-industries/community#1703]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

- Create an empty `.heex` file
- Paste the following
```
  <% academy_timezone = @conn.assigns.current_academy.timezone %>
     <%= if academy_timezone != @ts_timezone do %>
```

- on next key press zed crashes

![zed crash](https://github.com/zed-industries/community/assets/85486707/40f01071-570b-4617-9eb1-fa4514c56f3f)

### Environment

Zed: v0.91.3 (stable)
OS: macOS 13.4.0
Memory: 16 GiB
Architecture: x86_64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

[zed.log](https://github.com/zed-industries/community/files/11870111/zed.log)

**Comment by JosephTLyons** [comment]
Thanks for the report @ArthurMmn, we'll take a look into this.

**Comment by maxbrunsfeld** [comment]
I really appreciate the minimal repro, @ArthurMmn. I saw those crashes coming in, and was hoping somebody would figure out a small example like that. We'll take care of this ASAP.

**Comment by maxbrunsfeld** [comment]
I believe this is fixed in Zed 0.91.4. Let me  know if you are still having problems with HEEx files after upgrading; if so I'll reopen.
